src/PlaywrightSharp.Tests/Launcher/UserDataDirTests.cs

```diff
@@ -70,21 +72,25 @@ public async Task UserDataDirOptionShouldRestoreState()
         {
             using var userDataDir = new TempDirectory();
             var options = TestConstants.GetDefaultBrowserOptions();
-            options.Args = options.Args.Concat(new[] { $"--user-data-dir=\"{userDataDir}\"" }).ToArray();
+            options.UserDataDir = userDataDir.Path;
 
-            using (var browser = await Playwright.LaunchAsync(options))
-            {
-                var page = await browser.DefaultContext.NewPageAsync();
-                await page.GoToAsync(TestConstants.EmptyPage);
-                await page.EvaluateAsync("localStorage.hey = 'hello'");
-            }
+            var browser = await Playwright.LaunchAsync(options);
 
-            using (var browser2 = await Playwright.LaunchAsync(options))
-            {
-                var page2 = await browser2.DefaultContext.NewPageAsync();
-                await page2.GoToAsync(TestConstants.EmptyPage);
-                Assert.Equal("hello", await page2.EvaluateAsync<string>("localStorage.hey"));
-            }
+            var page = await browser.DefaultContext.NewPageAsync();
+            await page.GoToAsync(TestConstants.EmptyPage);
+            await page.EvaluateAsync("localStorage.hey = 'hello'");
+            await browser.CloseAsync();

```

Comment:
Dispose calls are fire and forget, we need to await for the close here.

Reply:
maybe we should implement `IAsyncDisposable`?

Reply:
The issue with IAsyncDisposable is that it's .NET standard 2.1
That would leave .NET Framework out. I will try to see if I can make it multi-target in another PR.